Source/drlg_l1.cpp

```diff
@@ -1416,9 +1416,9 @@ static void L5firstRoom()
 		ys = 1;
 		ye = DMAXY - 1;
 
-		VR1 = GenerateRnd(2);
-		VR2 = GenerateRnd(2);
-		VR3 = GenerateRnd(2);
+		VR1 = (GenerateRnd(2) != 0);
+		VR2 = (GenerateRnd(2) != 0);
+		VR3 = (GenerateRnd(2) != 0);
 
 		if (VR1 + VR3 <= 1)
 			VR2 = true;

```

Comment:
This is a bit of a weird one.

VR* is a bool. So yes, definitely makes sense to assign comparison result (true/false) to them.

But a few lines below, they are again used like arithmetic types, and added together.

I'd change ```VR1 + VR3 <= 1``` to ```!(VR1 && VR2)```

Reply:
The original code didn't use proper bool types so they were sometimes "flexible" about the usage. Now at least we have a chance to correct some of this. Nice catch.